### PR TITLE
Issue#1699 template list

### PR DESCRIPTION
**Pull request by Hugo-Hontans**
- [x] Convert dates (timestamp to Date) of Template with akita store
- [x] Update view of template-list page
- [x] Add row link to filter-table

### CODE REVIEW

File: libs/material/src/lib/template/+state/template.firestore.ts [hunk 1]
```diff
@@ -2,11 +2,16 @@ import { firestore } from "firebase";
...
 
-/** Document model of a template. */
-export interface TemplateDocument {
+/** Raw model of a template. */
+export interface TemplateRaw<D> {
   id: string;
   name: string;
   orgId: string;
-  created: Timestamp;
+  created: D;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/+state/template.firestore.ts]: I thought this kind of information were in `_meta`. Like `createdBy` for example. Have you checked with Bruce ?

File: libs/material/src/lib/template/+state/template.model.ts [hunk 2]
```diff
@@ -10,7 +10,19 @@ export function createTemplate(template: Partial<Template>): Template {
...
   }
 }
+
+/** Convert an TemplateWithTimestamps to an Template (that uses Date). */
+export function convertTemplateWithTimestampsToTemplate(
+  template: TemplateWithTimestamps
+): Template {
+  return {
+    ...template,
+    created: (template.created instanceof Date) ? template.created : template.created.toDate(), // prevent error in case the guard is wrongly called twice in a row
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/+state/template.model.ts]: Check with Yohann I asked him to create a reusable function `toDate(date: Timestamp | Date)` that we can use everywhere

File: libs/material/src/lib/template/pages/template-list/template-list.component.html [hunk 3]
```diff
@@ -1,13 +1,22 @@
-<section>
-  <article fxLayout="row" fxLayoutAlign="space-between center">
-    <h2>Templates</h2>
-    <mat-divider fxFlex></mat-divider>
-    <a mat-stroked-button color="primary" (click)="addTemplateDialog()">
+<section *ngIf="(templates$ | async) as templates">
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/pages/template-list/template-list.component.html]: use `ng-container`. You don't need this section

File: libs/ui/src/lib/icon-component.ts [hunk 4]
```diff
@@ -238,7 +238,10 @@ const icons = [
   },
   { name: 'wallet', url: 'assets/icons/wallet.svg' },
   { name: 'world', url: 'assets/icons/world.svg' },
-  { name: 'wrench', url: 'assets/icons/wrench.svg' }
+  { name: 'wrench', url: 'assets/icons/wrench.svg' },
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/icon-component.ts]: Follow alphabetic order please or we won't be able to find our way in.

File: libs/ui/src/lib/icon-component.ts [hunk 5]
```diff
@@ -238,7 +238,10 @@ const icons = [
...
+  { name: 'profile', url: 'assets/icons/profil.svg' },
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/icon-component.ts]: you have a typo in the name of the file (profil.svg -> profileE.svg)

File: libs/ui/src/lib/list/table-filter/table-filter.component.html [hunk 6]
```diff
@@ -17,7 +17,12 @@
     </ng-container>
 
     <tr mat-header-row *matHeaderRowDef="displayedColumns; sticky: true"></tr>
-    <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+    <ng-template [ngIf]="link" [ngIfElse]="withoutLink">
+      <tr mat-row *matRowDef="let row; columns: displayedColumns" [routerLink]="[link, row.id]" style="cursor: pointer;"></tr>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/table-filter/table-filter.component.html]: Sooo it works only for element with id ?

Comment (Hugo-Hontans) [reply libs/ui/src/lib/list/table-filter/table-filter.component.html]: For now I think we will have no link in the tables that do not use the id

File: libs/material/src/lib/template/+state/template.firestore.ts (hunk 1 above)
Comment (Hugo-Hontans) [reply libs/material/src/lib/template/+state/template.firestore.ts]: I created an issue to manage it correctly: #1791

File: libs/material/src/lib/template/+state/template.model.ts (hunk 2 above)
Comment (Hugo-Hontans) [reply libs/material/src/lib/template/+state/template.model.ts]: This function is not yet merge, I add a comment

**Pull request merged**